enhancements/microshift/kubernetes-for-industrial-edge.md

```diff
@@ -0,0 +1,436 @@
+for configuring the operating system components and anything that runs
+on the device outside of MicroShift.
+
+Many of the OpenShift APIs are not relevant for edge device use
+cases. We are starting with the absolute minimal set for now because
+it will be easier to add more APIs later than it will be to remove
+anything. In addition to the standard APIs included in core
+Kubernetes, the OpenShift APIs supported will be:
+
+* route.openshift.io/v1 for Routes
+* security.openshift.io/v1 for SecurityContextConstraints
+* authorization.openshift.io/v1 for RBAC APIs such as Role,
+  RoleBinding, etc.
+
+In addition to those APIs, we will enable two OpenShift controllers
+related to Ingress:
+
+* openshift.io/ingress-ip
+* openshift.io/ingress-to-route
+
+MicroShift is built downstream of OpenShift. The versions of the
+embedded components are selected by looking at an OpenShift release
+image, finding the git SHAs for the direct dependencies, resolving the
+secondary dependencies, and vendoring the results into the MicroShift
+git repository. The version of the containerized components are taken
+from the same release image and the image references are built into
+the MicroShift binary.
+
+A great deal of effort has gone into making OpenShift 4 self-managing,
+especially when it comes to multi-node deployment and upgrades. We do
+not want to reinvent that work, nor do we want to recreate the
+OpenShift 3 model using an outside orchestration tool.  A single-node
+control plane becomes a single point of failure when worker nodes are
+attached, which eliminates the high availability benefits of running
+workloads on kubernetes. Those statements taken together lead us to
+conclude that MicroShift will only support deployments with 1 node
+acting as both control plane and worker.
+
+Because MicroShift does not reuse the system operators from OpenShift,
+it will not be configured through API resources in the way that
+OpenShift is. Instead, we will use a new configuration file for the
+control plane (`/etc/microshift/config.yaml`) with [limited
+configuration
+options](https://github.com/openshift/microshift/blob/main/pkg/config/config.go). The
+node-side configuration will be done via the standard kubelet
+configuration file.
+
+Other operating system services will be configured outside of
+MicroShift using their standard configuration files. There is no
+machine-config-operator.
+
+Because MicroShift will be treated as a single application, we will
+not expose separate health checks or implement per-component restart
+features. If any embedded component is unhealthy, the application will
+be treated as unhealthy and be restarted.
+
+### Workflow Description
+
+#### Deploying
+
+Application deployers will use RHEL for Edge tools such as Image
+Builder to construct full system images containing RHEL, MicroShift,
+agents, custom operating system configuration, data prerequisites, and
+optionally application images. Those system images will be installed
```

Comment:
Again, suggest to avoid the term "installed" as it implies "an installer", but instead talk about imaging a device or cloning system images to devices.

Reply:
updated